src/plugin/ipc/file/filewrappers.cpp

```diff
@@ -371,7 +371,15 @@ static int _open_open64_work(int(*fn) (const char *path, int flags, ...),
     newpath = currPtsDevName;
   }
 
-  int fd = (*fn) (newpath, flags, mode);
+  dmtcp::string hook_path = "";
+  bool doSwap = pathvirt_get_physical_path ? pathvirt_get_physical_path(newpath, hook_path) : false;
+  int fd = -1;
+
+  /* hook_path was NULL, not swapping */
+  if (!doSwap)
+    fd = (*fn)(newpath, flags, mode);
+  else
+    fd = (*fn)(hook_path.c_str(), flags, mode);
 
   if (fd >= 0 && dmtcp_is_running_state()) {
     FileConnList::instance().processFileConnection(fd, newpath, flags, mode);
```

Comment:
@rohgarg, feel free to comment further.

@mossberg, I believe the use of `newpath` is correct here.  The model is that `newpath` is the path seen prior to checkpoint.  This is considered the "virtual path", since it may be changed to point to a different physical path on restart.

Since the physical path can change many times (once for each restart), the virtual path becomes the canonical name of a file.  Since newpath is a virtual path, this is the one to use for the internal state of DMTCP.

(Stated differently, the internal state of DMTCP should always refer to the virtual path (the path prior to checkpoint).  Any system calls should convert between the physical path known to the filesystem and the virtual path used for the internal state of DMTCP.)

Also, please seem my next inline note, where I have a related suggestion to simplify and improve the readability of the code.